### ISSUE TITLE
MueLu: missing parameter in aggregation

### ISSUE THREAD

**Issue by mayrmt** [trilinos/Trilinos#8846]
## Bug Report
@trilinos/muelu 

### Description

When running `ctest -VV` on a current version of `develop` (0f69cc4a68000d6e4003faf00887e23791f4de06), this error occurs in `MueLu_Structured_Unstructured_Region_Example_Poisson3D_27pt_stencil_MPI_4`:

```
33: Test command: /usr/bin/mpiexec "--oversubscribe" "-np" "4" "/hdd/codes/mayrmt_trilinos/build_muelu/packages/muelu/research/regionMG/example/MueLu_RegionExampleDriver.exe" "--linAlgebra=Tpetra" "--xml=structured_unstructured_1dof.xml" "--matrixType=Brick3D" "--nx=28" "--ny=28" "--nz=28" "--smootherIts=2" "--unstructured={1,2}"
33: Test timeout computed to be: 1500
33: ========================================================
33: Xpetra::Parameters
33:  Linear algebra library: Tpetra
33: Galeri::Xpetra::Parameters<long>
33:  Matrix type: Brick3D
33:  Problem size: 21952
33: Processor subdomains in x direction: 2
33: Processor subdomains in y direction: 2
33: Processor subdomains in z direction: 1
33: ========================================================
33: Galeri complete.
33: ========================================================
33: nodesPerDim: {15, 14, 28}
33: nodesPerDim: {14, 15, 28}
33: Clearing old data (if any)
33: 
33: ******* WARNING *******
33: Hierarchy::ReplaceCoordinateMap: matrix and coordinates maps are same, skipping...
33: MueLu::Amesos2Smoother: using "Klu"
33: Using default factory (SmootherFactory[17] ) for building 'CoarseSolver'.
33: Using default factory (AmalgamationFactory[20] ) for building 'UnAmalgamationInfo'.
33: Using default factory (CoarseMapFactory[21] ) for building 'CoarseMap'.
33: Level 0
33: Level 1
33: Build (MueLu::TentativePFactory)
33: Build (MueLu::HybridAggregationFactory)
33: Build (MueLu::CoalesceDropFactory)
33: Build (MueLu::AmalgamationFactory)
33: AmalagamationFactory::Build(): found fullblocksize=1 and stridedblocksize=1 from strided maps. offset=0
33: lightweight wrap = 1
33: algorithm = "classical" classical algorithm = "default": threshold = 0, blocksize = 1
33: 
33: ******* WARNING *******
33: Level::Set: unable to store "Filtering" generated by factory CoalesceDropFactory[1](0x55588922dd30) on level 0, as it has not been requested and no keep flags were set for it
33: Detected 0 Dirichlet nodes
33: Number of dropped entries in unamalgamated matrix graph: 0/564898 (0%)
33: Algo "Aggretation: structured algorithm"
33: BuildAggregates (Aggretation: structured algorithm)
33: 
33: p=1: *** Caught standard std::exception of type 'Teuchos::Exceptions::InvalidParameterName' :
33: 
33:  Error!  The parameter "aggregation: phase2a agg factor" does not exist
33:  in the parameter (sub)list "ANONYMOUS".
33:  
33:  The current parameters set in (sub)list "ANONYMOUS" are:
33:  
33:    {
33:      "aggregation: max agg size" : int = 2147483647
33:      "aggregation: min agg size" : int = 2
33:      "aggregation: max selected neighbors" : int = 0
33:      "aggregation: ordering" : string = natural
33:      "aggregation: enable phase 1" : bool = 1
33:      "aggregation: enable phase 2a" : bool = 1
33:      "aggregation: enable phase 2b" : bool = 1
33:      "aggregation: enable phase 3" : bool = 1
33:      "aggregation: preserve Dirichlet points" : bool = 0
33:      "aggregation: allow user-specified singletons" : bool = 0
33:      "aggregation: error on nodes with no on-rank neighbors" : bool = 0
33:      "aggregation: phase2a include root" : bool = 1
33:      "aggregation: phase3 avoid singletons" : bool = 0
33:      "aggregation: coarsening rate" : string = {2}
33:      "aggregation: coarsening order" : int = 0
33:      "aggregation: number of spatial dimensions" : int = 3
33:      "aggregation: use interface aggregation" : bool = 1
33:      "Graph" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0x558c308b5100,node=0x558c30d89e80,strong_count=5,weak_count=0}
33:      "DofsPerNode" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0x558c308b5100,node=0x558c30d89e80,strong_count=5,weak_count=0}
33:      "OnePt aggregate map name" : string = 
33:      "OnePt aggregate map factory" : string = 
33:      "Interface aggregate map name" : string = 
33:      "Interface aggregate map factory" : string = 
33:      "interfacesDimensions" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0,node=0,strong_count=0,weak_count=0}
33:      "nodeOnInterface" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0,node=0,strong_count=0,weak_count=0}
33:      "numDimensions" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0,node=0,strong_count=0,weak_count=0}
33:      "lNodesPerDim" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0,node=0,strong_count=0,weak_count=0}
33:      "aggregationRegionType" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0,node=0,strong_count=0,weak_count=0}
33:    }
33:  
33:  
33:  Throw number = 1
33:  
33:  
33: 
33: p=2: *** Caught standard std::exception of type 'Teuchos::Exceptions::InvalidParameterName' :
33: 
33:  Error!  The parameter "aggregation: phase2a agg factor" does not exist
33:  in the parameter (sub)list "ANONYMOUS".
33:  
33:  The current parameters set in (sub)list "ANONYMOUS" are:
33:  
33:    {
33:      "aggregation: max agg size" : int = 2147483647
33:      "aggregation: min agg size" : int = 2
33:      "aggregation: max selected neighbors" : int = 0
33:      "aggregation: ordering" : string = natural
33:      "aggregation: enable phase 1" : bool = 1
33:      "aggregation: enable phase 2a" : bool = 1
33:      "aggregation: enable phase 2b" : bool = 1
33:      "aggregation: enable phase 3" : bool = 1
33:      "aggregation: preserve Dirichlet points" : bool = 0
33:      "aggregation: allow user-specified singletons" : bool = 0
33:      "aggregation: error on nodes with no on-rank neighbors" : bool = 0
33:      "aggregation: phase2a include root" : bool = 1
33:      "aggregation: phase3 avoid singletons" : bool = 0
33:      "aggregation: coarsening rate" : string = {2}
33:      "aggregation: coarsening order" : int = 0
33:      "aggregation: number of spatial dimensions" : int = 3
33:      "aggregation: use interface aggregation" : bool = 1
33:      "Graph" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0x55701e0751a0,node=0x55701e549090,strong_count=5,weak_count=0}
33:      "DofsPerNode" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0x55701e0751a0,node=0x55701e549090,strong_count=5,weak_count=0}
33:      "OnePt aggregate map name" : string = 
33:      "OnePt aggregate map factory" : string = 
33:      "Interface aggregate map name" : string = 
33:      "Interface aggregate map factory" : string = 
33:      "interfacesDimensions" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0,node=0,strong_count=0,weak_count=0}
33:      "nodeOnInterface" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0,node=0,strong_count=0,weak_count=0}
33:      "numDimensions" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0,node=0,strong_count=0,weak_count=0}
33:      "lNodesPerDim" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0,node=0,strong_count=0,weak_count=0}
33:      "aggregationRegionType" : Teuchos::RCP<MueLu::FactoryBase const> = Teuchos::RCP<MueLu::FactoryBase const>{ptr=0,node=0,strong_count=0,weak_count=0}
33:    }
33:  
33:  
33:  Throw number = 1
```

### Steps to Reproduce
1. SHA1: 0f69cc4a68000d6e4003faf00887e23791f4de06
1. Configure script: 
[do-configure-trilinos_muelu.txt](https://github.com/trilinos/Trilinos/files/6098156/do-configure-trilinos_muelu.txt)
1. Configure log:  [attach here]
1. Build log:  [attach here]
1. Input deck:  [attach here]
1. Run `ctest -VV` for all tests or `ctest -VV MueLu_Structured_Unstructured_Region_Example_Poisson3D_27pt_stencil_MPI_4` for this particular test.
1. Run log:  Error message see above

## Related Issues

- Blocks PR #8506

**Comment by mayrmt** [comment]
Looks like an issue with the result of #8828. 

@csiefer2, any ideas?

**Comment by mayrmt** [comment]
I can verify that this has been fixed by @pohm01 via PR #8860. Closing.
